src/main/java/org/generationcp/middleware/dao/dms/StockDao.java

```diff
@@ -292,5 +305,119 @@ public void replaceExperimentStocks(final Integer oldStockId, final Integer newS
+			query.addScalar("entryNumber", new IntegerType());
+			query.addScalar("gid", new IntegerType());
+			query.addScalar("designation", new StringType());
+			query.addScalar("entryCode", new StringType());
+			query.addScalar("lots", new IntegerType());
+			query.addScalar("available", new StringType());
+			query.addScalar("unit", new StringType());
+			for (final MeasurementVariable entryDescriptor : studyEntrySearchDto.getEntryDescriptors()) {
+				final String entryName = entryDescriptor.getName();
+				query.addScalar(entryName + "_propertyId", new IntegerType());
+				query.addScalar(entryName + "_variableId", new IntegerType());
+				query.addScalar(entryName + "_value", new StringType());
+			}
+
+			query.setParameter("studyId", studyEntrySearchDto.getStudyId());
+			if (studyEntrySearchDto.getFilter()!=null) {
+				if (studyEntrySearchDto.getFilter().getEntryNumbers()!=null && !studyEntrySearchDto.getFilter().getEntryNumbers().isEmpty()) {
+					query.setParameterList("entryNumbers", studyEntrySearchDto.getFilter().getEntryNumbers());
+				}
+				if (studyEntrySearchDto.getFilter().getEntryIds()!=null && !studyEntrySearchDto.getFilter().getEntryIds().isEmpty()) {
+					query.setParameterList("entryIds", studyEntrySearchDto.getFilter().getEntryIds());
+				}
+			}
+			query.setResultTransformer(AliasToEntityMapResultTransformer.INSTANCE);
+			GenericDAO.addPaginationToSQLQuery(query, pageable);
+
+			final List<Map<String, Object>> results = query.list();
+			final List<StudyEntryDto> studyEntryDtos = new ArrayList<>();
+			for (final Map<String, Object> row : results) {
+				final StudyEntryDto studyEntryDto = new StudyEntryDto();
+				studyEntryDto.setEntryId((Integer) row.get("entryId"));
+				studyEntryDto.setGid((Integer) row.get("gid"));
+				studyEntryDto.setEntryNumber((Integer) row.get("entryNumber"));
+				studyEntryDto.setDesignation((String) row.get("designation"));
+				studyEntryDto.setEntryCode((String) row.get("entryCode"));
+				studyEntryDto.setLots((Integer) row.get("lots"));
+				studyEntryDto.setAvailable((String) row.get("available"));
+				studyEntryDto.setUnit((String) row.get("unit"));
+				final Map<String, StudyEntryPropertyData> variables = new HashMap<>();
+				for (final MeasurementVariable entryDescriptor : studyEntrySearchDto.getEntryDescriptors()) {
+					final StudyEntryPropertyData studyEntryPropertyData =
+						new StudyEntryPropertyData((Integer) row.get(entryDescriptor.getName() + "_propertyId"),
+							(Integer) row.get(entryDescriptor.getName() + "_variableId"),
+							(String) row.get(entryDescriptor.getName() + "_value"));
+					variables.put(entryDescriptor.getName(), studyEntryPropertyData);
+				}
+				//These elements should not be listed as germplasm descriptors, this is a way to match values between column
+				//and table cells. In the near future this block should be removed
+				variables.put("GID", new StudyEntryPropertyData(String.valueOf(studyEntryDto.getGid())));
```

Comment:
We are hardcoding column names, how would this behave if an alias was used for these variables (eg. through Data Import or Import Design)?